openupgrade_scripts/scripts/purchase_stock/18.0.1.2/pre-migration.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2025 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_new_columns = [
+    ("purchase.order.line", "group_id", "many2one"),
+    ("purchase.order.line", "location_final_id", "many2one"),
+    ("stock.warehouse.orderpoint", "product_supplier_id", "many2one"),
+]
+
+
+def fill_purchase_order_line_group_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE purchase_order_line pol
+        SET group_id = sr.group_id
+        FROM stock_move sm
+        JOIN stock_rule sr ON sm.rule_id = sr.id AND
+            sr.group_propagation_option = 'propagate'
+        WHERE sm.purchase_line_id = pol.id AND sr.group_id IS NOT NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE purchase_order_line pol
+        SET group_id = po.group_id
+        FROM purchase_order po
+        JOIN stock_rule sr ON po.group_id = sr.group_id AND
+            sr.group_propagation_option = 'propagate'
+        WHERE pol.order_id = po.id AND pol.group_id IS NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE purchase_order_line pol
+        SET group_id = pg.id
+        FROM sale_order_line sol
+        JOIN procurement_group pg ON pg.sale_id = sol.order_id
+        JOIN stock_rule sr ON pg.id = sr.group_id AND
+            sr.group_propagation_option = 'propagate'
+        WHERE pol.sale_line_id = sol.id AND pol.group_id IS NULL
+        """,
+    )
+
+
+def fill_purchase_order_line_location_final_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE purchase_order_line pol
+        SET location_final_id = sm.location_dest_id
```

Comment:
this is not what you say you do below, I think the work file is right

Reply:
Yes, it's a small slip.